---
title: Phylogenetic Treatment of Recombination
reveal:
    center: false
    history: true
    slideNumber: true
    theme: theme.css
---

<section class="titlepage center">
  <div class="title">Phylogenetic Treatment of Recombination</div>
  <div class="subtitle">With a particular focus on Bacterial Recombination</div>
  <div class="authors">Tim Vaughan</div>

  <div class="institution">cEvo group, D-BSSE, The ETH Zurich</div>

  <div class="date">Taming the BEAST, 22<sup>nd</sup> June, 2018</div>

  <div>
    <img data-src="Taming-the-BEAST-Oberageri.png" style="width:200px;box-shadow:none;margin-top:5%">
  <img data-src="cEvo_logo.png" style="width:200px;box-shadow:none;margin-top:5%">
  <img data-src="eth_logo.png" style="width:200px;box-shadow:none;margin-top:5%">
  </div>

</section>

<!-- What is recombination? -->

<section>
  <h1>Recombination primer</h1>

  <div style="display:inline-block;width:60%;vertical-align:middle">
  <ul class="spaced" style="width:80%">
    <li>Occurs in eukaryotes, bacteria and viruses.</li>
    <li>Eukaryotic recombination occurs during meiosis via chromosomal crossover.</li>
    <li>Bacterial recombination occurs via
      <ul>
        <li>phage-mediated transduction,</li>
        <li>natural transformation, and</li>
        <li>conjugation.</li>
      </ul>
    </li>
    <li>Viral recombination occurs when multiple strains infect a single
      cell. (Example: reassortment in segmented viral genomes.)</li>
  </ul>
  </div>
  <div style="display:inline-block;width:39%;vertical-align:middle">
    <img data-src="HR_meiosis.png" style="width:100%">
    <div class="cite">commons.wikimedia.org</div>
  </div>
</section>

<section>
  <h1>Recombination and Phylogenetics</h1>

  <img data-src="local_trees.svg" style="width:100%">

  <p>Ignoring recombination when present:</p>
  <ul>
    <li>Biases inference of internal node heights times.</li>
    <li>Biases inference of topology.</li>
    <li>Poor mixing due to badly fitting model.</li>
  </ul>
</section>

<section>
  <h1>Relationship to Species Phylogenies</h1>

  <div style="display:inline-block;width:49%">
    <img data-src="MSC_norecomb.svg" style="width:100%">
  </div>
  <div style="display:inline-block;width:49%">
    <img data-src="MSC_withrecomb.svg" style="width:100%" class="fragment">
  </div>

  <ul class="fragment">
    <li>Recombination affects gene phylogenies.</li>
    <li>Species phylogeny may also be a network! (See <a class="cite" href="https://doi.org/10.1093/molbev/msx307">Zhang et al., MBE, 2017</a>.)</li>
  </ul>
</section>

<section>
  <h1>Accounting for Recombination</h1>

    <ol class="spaced">
        <li>Pre-process of data to identify and remove
            non-vertically inherited material. 

            <table style="width:800px">
                <tr><th style="width:400px">Pros</th> <th>Cons</th></tr>
                <tr>
                    <td>
                        <ul>
                            <li>Can use standard tools for
                                phylogenetic inference.</li>
                        </ul>
                    </td>
                    <td>
                        <ul>
                            <li>Data is being thrown away.</li>
                            <li>Ad hoc, may bias results.</li>
                        </ul>
                    </td>
                </tr>

            </table>
        </li>

        <li class="fragment" style="margin-top:20px">Explicitly model recombination processes.<br>
            <table style="width:800px">
                <tr><th style="width:400px">Pros</th> <th>Cons</th></tr>
                <tr>
                    <td>
                        <ul>
                            <li>Can make use of all data.</li>
                            <li>Can infer additional parameters such as recombination rates.</li>
                            <li>May yield increased confidence in estimates.</li>
                        </ul>
                    </td>
                    <td>
                        <ul>
                            <li>Models can be complex, with many parameters.</li>
                            <li>Both computationally and statistically challenging.</li>
                        </ul>
                    </td>
                </tr>
            </table>
        </li>
    </ul>
</section>

<!--section class="center">
  <div style="font-weight:600;font-size:3em">CHALLENGE&nbsp;&nbsp;&nbsp;ACCEPTED</div>
</section-->

<section>
  <h1>Wright-Fisher with Recombination</h1>

    <ul>
        <li>Consider an WF model with male and female diploid individuals.</li>
        <li>Focus on a small segment of the genome.</li>
    </ul>

    <img data-src="WFrec.svg" style="width:100%">

    <ul>
        <li>Each child selects 1 male and 1 female parent uniformly at random.</li>
        <li>With probability $r$ (which depends on the segment size) the two
            chromosome pairs of a parent are recombined before being passed
            on.</li>
    </ul>
</section>

<section>
  <h1>Wright-Fisher model with recombination</h1>

  <ul>
    <li>Since the specific pairing of chromosomes only matters over a
      single generation, in the long term the haploid approximation is
      very good:</li>
  </ul>

  <img data-src="WFrec_chromosome.svg" style="width:100%;margin-top:2em">

  <ul style="margin-top:1em">
    <li>Each child in $i+1$ selects a parent uniformly at random from
      generation $i$.</li>
    <li>With probability $r$ an additional parent is selected.</li>
    <li>In this case, a breakpoint chosen uniformly on the chromosome, and
      everything to either the left or right replaced by the homologous
      section of the second parent's chromosome.</li>
    <li>(Equivalent to selecting 2 parents for each child and letting the
      resulting sequences recombine with some probability.)
  </ul>
</section>

<section>
  <h1>Model 1: Coalescent with Recombination</h1>

      <div style="width:55%;display:inline-block;vertical-align:middle">
        <p>For fixed recombination rate $\rho=r/g$ in the limit $r\ll 1$, $g\ll 1$
        and $N\gg 1$, the genealogical process is the coalescent with
        recombination (<a class="cite" href="https://doi.org/10.1016/0040-5809(83)90013-8">Hudson, 1983</a>):</p>

        <ul>
            <li>Coalescence rate: $T_c(k)=\binom{k}{2}\frac{1}{N}$</li>
            <li>Recombination rate: $T_r(k)=\rho k$.</li>
            <li>Recombination break points chosen uniformly along sequence: everything
            to the left descends from one parent, everything else to the other.</li>
            <li>Each site possess a <b>local tree</b>.</li>
            <li>Local trees may find MRCAs (grey sites) before <b>grand (G)MRCA</b> of the process.</li>
        </ul>
    </div>
    <img data-src="CWR.svg" style="width:40%;display:inline-block;vertical-align:middle">
</section>

<section>
    <h1>CwR example ARG</h1>

    <img data-src="simulated_arg.png" style="width:100%">

</section>

<section>
    <h1>Coalescent with gene conversion</h1>
    <div style="width:55%;display:inline-block;vertical-align:middle">
        <p>A gene conversion event refers to the replacement of a single
        contiguous stretch of sequence with a homologous stretch from a
        different parent. Incorporated into a modified CwR process by
        <a class="cite" href="http://www.genetics.org/content/155/1/451.short"> Wiuf&nbsp;and&nbsp;Hein (2000)</a>.</p>
        <ul>
            <li>Allows similar patterns of site ancestry but with fewer events.</li>
            <li>Model applicable to prokaryotic recombination.</li>
            <li>Coalescence and recombination rates as for CwR.</li>
            <li>Conversions initiate at randomly chosen starting sites and
            extend for $d$ sites where
            $P(d)=(1-\delta^{-1})^{d-1}\delta^{-1}$ (geometric distribution).</li>
        </ul>

    </div>
    <img data-src="CWGC.svg" style="width:40%;display:inline-block;vertical-align:middle">

</section>

<section>
    <h1>Bayesian inference of ARGs</h1>

    <p>Can easily write down an expression for the posterior distribution for the  ARG given an alignment:</p>
    <blockquote>
        $$P(G,\rho,N,\mu|A) = \frac{1}{P(A)}P(A|G,\mu)P(G|\rho,N)P(\rho,N,\mu)$$
    </blockquote>
    <ul>
        <li>$G$ is the recombination graph,</li>
        <li>$\rho$ is the recombination rate.</li>
    </ul>

    <p>In practice, this is non-trivial since:</p>
    <ol>
        <li>the likelihood for $G$, $P(A|G,\mu)$, is invariant under many features of $G$,</li>
        <li>the likelihood surface often contains many distinct peaks, and</li>
        <li>the state space of ARGs is <b>huge</b>: much larger than for trees.</li>
    </ol>

    <p style="margin-top:1em">Despite this, many approximate algorithms exist.</p>
</section>

<section>
    <h1>Algorithms for Bayesian ARG inference</h1>

    <ul class="spaced" style="margin-top:10%">
        <li>SMARTIE (<a class="cite" href="https://doi.org/10.1093/sysbio/syp076">Bloomquist and Suchard, 2010</a>)</li>
        <li>ARGweaver (<a class="cite" href="http://dx.doi.org/10.1371/journal.pgen.1004342">Rasmussen et al, 2014</a>)</li>
        <li>ClonalFrame (<a class="cite" href="https://dx.doi.org/10.1534/genetics.106.063305">Didelot and Falush, 2007</a>)</li>
        <li>ClonalOrigin (<a class="cite" href="https://dx.doi.org/10.1534/genetics.110.120121">Didelot et al., 2010</a>)</li>
    </ul>
</section>

<section>
    <h1>Sequentially Markov Coalescent (SMC)</h1>
    <ul>
        <li><a class="cite" href="https://doi.org/10.1006/tpbi.1998.1403">Wiuf
                and Hein (1999)</a> reformulated the CwR as a "spatial" process
                            along sequences.</li>
        <li><a class="cite"
        href="http://dx.doi.org/10.1098/rstb.2005.1673">McVean and Cardin
        (2005)</a> modified this to produce a process for local trees which is
    Markovian along sequences.</li>
    </ul>

    <div class="figure">
        <img data-src="smc_figure.svg" style="width:100%;padding:1em;">
        <div class="cite" style="font-size:0.8em;text-align:right">
            <a class="cite" href="http://dx.doi.org/10.1098/rstb.2005.1673">McVean and Cardin (2005)</a>
        </div>
    </div>

    <ul>
        <li>Sites between breakpoints exponentially distributed with rate $\rho T$ where $T$ is the total length of the current local tree.</li>
        <li>Neglects some possible recombinations, e.g. those that do not affect the data.</li>
    </ul>
</section>

<section>
    <h1>Population inference using SMC</h1>

    <ul>
        <li><a class="cite" href="http://dx.doi.org/10.1038/nature10231">Li &amp; Durbin (2011)</a> developed an SMC-based HMM on pairs of alignments.</li>
        <li>Baum-Welch used to estimate parameters and hidden states (local TMRCAs).</li>
        <li>TMRCA distribution used to produce estimates of population size dynamics.</li>
    </ul>

    <div class="figure">
        <img data-src="li_durbin.jpg" style="height:400px;padding:1em;vertical-align:middle">
        <img data-src="li_durbin2.jpg" style="height:400px;padding:1em;vertical-align:bottom;margin-left:2em">
        <div class="cite" style="text-align:right;margin-top:1em"><a class="cite" href="http://dx.doi.org/10.1038/nature10231">Li and Durbin (2011)</a></div>
    </div>
</section>

<section>
    <h1>Bacteria</h1>

    <div style="float:left; width:700px">
        <ul>
            <li>Bacteria play critical roles in the health of animals
            and plants.</li>
            <li>Many bacteria possess interesting
                and <i>experimentally accessible</i> evolutionary
                dynamics.</li>
            <li>Bacterial genomes are measurably evolving over
                relatively short study periods.
                <div class="figure" style="width:500px; position: relative; left: 100px">
                    <img data-src="MEP_cropped.png">
                    <div class="cite">Drummond &amp; Rambaut, TIEE (2003)</div>
                </div>
            </li>
        </ul>
    </div>
    <div style="float:right; width:250px">
        <div class="figure">
            <img data-src="EColiGut_Psa_collage.jpg">
        </div>
        <div class="figure">
            <img data-src="adaptive_radiation.jpg">
            <div class="cite">Rainey &amp; Travisano, Nature (1998)</div>
        </div>
    </div>
</section>

<section>
    <h1>Cartoon bacterial model</h1>

    <div style="position: relative; top:0; left:0; width:1000px; height:400px">
        <img data-src="binary_fission_7.svg" class="fragment current-visible" data-fragment-index="1"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_6.svg" class="fragment" data-fragment-index="2"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_5.svg" class="fragment" data-fragment-index="3"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_4.svg" class="fragment" data-fragment-index="4"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_3.svg" class="fragment" data-fragment-index="5"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_2.svg" class="fragment" data-fragment-index="6"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_1.svg" class="fragment" data-fragment-index="7"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
    </div>

    <img data-src="transduction_cropped.jpg" class="fragment" data-fragment-index="5"
                                                              style="position:absolute; top:100px; left:0px; height:250px;">

    <img data-src="conjugation.jpg" class="fragment" data-fragment-index="7"
                                                     style="position:absolute; top:100px; right:0px; height:250px;">
    <ul class="fragment">
        <li>Effect of events can be plasmid transfer, insertion or homologous recombination.</li>
        <li>Rate of recombination varies and is subject to selection.</li>
        <li>Focus on homologous recombination: only event which doesn't alter the length of the sequence.</li>
    </ul>
</section>

<section>
    <h1>Bacterial Recombination Rates</h1>

    <p>For many bacteria, the ratio between the recombination rate and the mutation rate is extremely high.</p>

    <div class="figure">
        <img data-src="r2mhist.svg" style="width:800px">
        <div class="cite">Vos and Didelot, The ISME Journal (2009)</div>
    </div>
</section>

<section>
    <h1>The Coalescent with Gene Conversion</h1>

    <div style="position: relative; top:0; left:0; width:1000px; height:500px;">
        <!--img data-src="WrightFisherGC1.svg"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC2.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC3.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC3b.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC4b.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;"-->
        <img data-src="WrightFisherGC5b.svg"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
    </div>
    <a class="cite" href="">Wiuf, 1999</a>; <a class="cite" href="">Wiuf and Hein, 2000</a>

    <div style="position:absolute; top:100px; left:0px; width:350px">
        <table class="params">
            <tr><th colspan="2">Parameters</th></tr>
            <tr> <td>$1/N(t)$</td> <td>Coalescence rate</td> </tr>
            <tr> <td>$\rho_s$</td> <td>Recombination rate</td> </tr>
            <tr> <td>$\delta$</td> <td>Expected tract length</td> </tr>
        </table>
    </div>
</section>

<section class="center">
    <div class="textbox alert">
        <div class="title">Problem</div>
        The space of possible ancestral recombination graphs is extremely
        large. Even two samples have infinitely many distinct ancestries!
    </div>
</section>

<section>
    <h1>Full ARG</h1>

    <img data-src="full_arg.svg" style="box-shadow:none;width:500px">

    <br>
    &nbsp;
</section>

<section>
    <h1>Approximation 1: ClonalFrame</h1>

    <img data-src="CF_arg.svg" style="box-shadow:none;width:500px">

    <br>

    <a class="cite" href="">Didelot et al., 2007 (ClonalFrame)</a>;
    <a class="cite" href="">Didelot and Wilson, 2015 (ClonalFrameML)</a>
</section>

<section>
    <h1>Approximation 2: ClonalOrigin</h1>

    <img data-src="CO_arg.svg" style="box-shadow:none;width:500px">

    <br>

    <a class="cite" href="">Didelot et al., 2010 (ClonalOrigin)</a>;
    <a class="cite" href="">Ansari and Didelot, 2014 (ClonalOrigin')</a>
</section>

<section class="center">
    <div class="textbox" style="width:600px;position:relative;left:200px">
        <div class="title">Our Goal</div>
        To conduct Bayesian inference of the full ARG under the ClonalOrigin
        model, optionally under non-trivial population size dynamics.
    </div>
</section>


<section>
    <h1>Inference under the ClonalOrigin Model</h1>

    <p> Inference follows the standard Bayesian phylogenetic tradition: </p>

    \begin{equation*}
        f(G,N,\mu,\rho,\delta|A) \propto P_{F}(A|G,\mu)f_{CGC}(G|N,\rho,\delta)f_{\text{prior}}(N
,\mu,\rho,\delta)
    \end{equation*}

    <p>where</p>

    <ul>
        <li>$A$ is the sequence alignment,</li>
        <li>$\mu$ are the substitution model parameters, and</li>
        <li>$G$ is the full sample genealogy including clonal frame<br>
            $T$ and $M$ conversions $\{C_i\}_{i\in[1\ldots M]}$.</li>
    </ul>

    <div>
    <p>The genealogy density under ClonalOrigin model can be expanded</p>
    \begin{equation*}
        f_{CGC}(G|\rho',\delta,N)=\left(\prod_{i=1}^M f(C_i|T,N,\delta)\right)P(M|T,\rho)f_C(T|N)
    \end{equation*}
    </div>
</section>

<section>
    <h1>Identifiability</h1>

    <p>Despite using a simplified model, an infinite number of ARGs still
    possess the same likelihood given a sequence alignment.</p>

    <img data-src="identifiability.svg" style="box-shadow:none;width:800px">

    <blockquote>
        Very important for an MCMC algorithm to propose state changes which minimize effect on likelihood.
    </blockquote>
</section>

<section>
    <h1>MCMC State Proposal Strategy</h1>

    <div style="position: relative; top:0; left:0; width:1000px; height:500px;">
        <img data-src="collapse_expand1.svg"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
    </div>

    <ul>
        <li>Attachment points and site tracts of new conversions drawn from prior.</li>
    </ul>
</section>

<section>
    <h1>Validation 1: Sampling from the prior</h1>

    <ul>
        <li> It is trivial to sample directly from the approximate ARG prior
            via coalescent simulation.</li>
        <li> We expect precise agreement between our MCMC algorithm and the
            coalescent simulation when no data is involved.</li>
    </ul>

    <div class="fragment">
        <img data-src="validate_CFheight.svg" style="box-shadow:none">
        <img data-src="validate_meanEdgeLength.svg" style="box-shadow:none">
        <img data-src="validate_meanTractLength.svg" style="box-shadow:none">
    </div>

    <blockquote class="alert fragment">
        One should always test that a new sampler exactly converges to a known
        distribution: this provides a strict necessary criterion for
        correctness.
    </blockquote>
</section>

<section>
    <h1>Validation 2: Inference from simulated data</h1>

    <img data-src="posterior_rho.svg" style="box-shadow:none; padding:0; margin:0">
    <img data-src="posterior_delta.svg" style="box-shadow:none;padding:0;margin:0">
    <img data-src="posterior_theta.svg" style="box-shadow:none;padding:0;margin:0">
</section>

<section>
    <p style="margin:0px">True ARG:</p>
    <img data-src="true_tree.svg" style="box-shadow:none; padding:0; margin:0; width:800px">

    <div>
        <p style="margin:0px">Randomly-selected ARG from MCMC:</p>
        <img data-src="sampled_tree.svg" style="box-shadow:none; padding:0; margin:0; width:800px">
    </div>
</section>

<section>
    <h1>Summary networks</h1>

    <div style="position: relative; top:0; left:0; width:1000px; height:500px;">
        <img data-src="summarizing6.svg"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing5.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing3.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing2.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing1.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="summarizing0.svg" class="fragment"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
    </div>
</section>

<section>
    <p style="margin:0px">True ARG:</p>
    <img data-src="true_tree.svg" style="box-shadow:none; padding:0; margin:0; width:800px">

    <div class="fragment">
        <p style="margin:0px">Summary ARG from MCMC:</p>
        <img data-src="summary_tree.svg" style="box-shadow:none; padding:0; margin:0; width:800px">
    </div>
</section>

<section>
    <h1>Application to <i>Escherichia coli</i></h1>

    <ul>
        <li>Applied method to <i>E. coli</i> O157/O26 sequences collected by Tessy George (<sup>m</sup>EpiLab, Massey University, New Zealand).</li>
        <li>Sequences for 53 rMLST genes (21.1 kb total) from each of 23 human and cattle-derived isolates.</li>
        <li>Analyzed using fixed expected tract length parameter and informative prior on conversion rate parameter.</li>
    </ul>

    <img data-src="Tessy_rMLST_splitstree.svg" style="width:900px;box-shadow:none">
</section>

<section>
    <h1>Application to <i>Escherichia coli</i></h1>

    <div style="position: relative; top:0; left:0; width:1000px; height:600px;">
        <img data-src="Tessy_new_summary2.svg" style="position:absolute;top:0;left:0;width:1000px;box-shadow:none;background-color:white">
        <img class="fragment" data-src="Tessy_new_summary2_compare.svg" style="position:absolute;top:0;left:0;width:1000px;box-shadow:none;background-color:white">
        <img class="fragment" data-src="gene_flow.svg" style="position:absolute;top:0;left:0;width:250px;background-color:white;padding:20px">
    </div>
</section>

<section>
    <h1>Software implementation</h1>

    <img data-src="bacter_logo.svg" style="width:500px;box-shadow:none">

    <br>

    <a href="http://tgvaughan.github.io/bacter">tgvaughan.github.io/bacter</a>

    <br>
    <br>

    <ul>
        <li>BEAST 2 package that performs inference under the ClonalOrigin model.</li>
        <li>Joint inference of the clonal frame, recombinant edges and parameters.</li>
        <li>Can be combined with usual variety of substitution models, parametric population models and parameter priors.</li>
        <li>GUI analysis configuration via BEAUti.</li>
    </ul>
</section>


<section>
    <h1>Bayesian Skyline Plots in Bacter</h1>

    <img data-src="skyline_icon.svg" style="box-shadow:none;width:400px">

    <ul>
        <li>Bacter implements a non-parametric demographic history model inspired by the standard Bayesian Skyline Plot <a href="https://doi.org/10.1093/molbev/msi103">(Drummond et al., 2005)</a>.</li>
        <li>In principle can reconstruct non-trivial dynamics from very few genomes.</li>
        <li>In practice, this is limited by the computational expense of the algorithm.</li>
    </ul>
</section>

<section>
    <h1>Piecewise constant/linear population model</h1>

    <div style="position: relative; top:0; left:0; width:1000px; height:600px;">
        <img data-src="skyline_diagram.svg" style="position:absolute;left:0;top:0;width:1000px;box-shadow:none">
        <img class="fragment" data-src="skyline_linear_diagram.svg" style="position:absolute;left:0;top:0;width:1000px;box-shadow:none;background-color:white">
    </div>
</section>

<section>
    <h1>Inference example</h1>

    <img data-src="skyline_tree.svg" style="width:700px;box-shadow:none;margin:0">


    <img class="fragment" data-src="popfun_inf.svg" style="width:800px;margin-top:0">
</section>

<section>
     <h1>Bacter limitations</h1>

     <div style="float:left;width:600px">
     <ul>
         <li>Computational complexity scales with the number of proposed conversions.
             <ul>
                 <li>This can be arbitrarily large, even for small sample sets.</li>
                 <li>Cannot use the BEAGLE phylogenetic likelihood library to speed things up due
                     to peculiarities of ARG likelihood computation.</li>
             </ul>
             <br>
         </li>
         <li>MCMC algorithm used does not intelligently locate conversions.
             <ul>
                 <li>Looking at addressing this in the near future.</li>
             </ul>
             <br>
         </li>

         <li>Network summary algorithm can produce peculiar results.
             <ul>
                 <li>More research needs to be done to find a better algorithm.</li>
             </ul>
         </li>
     </ul>
     </div>
     <div style="float:right">
         <br>
         <br>
         <img data-src="ointment.svg" style="width:300px;box-shadow:none;"/>
     </div>
</section>

<section>
  <h1>Summary</h1>

  <ol class="spaced" style="margin-top:5%">

    <li>It is important to be aware of the role recombination plays in shaping the phylogeny of a set of samples.</li>
    <li>Recombination networks are distinct from (nested within!) species networks.</li>
    <li>Network inference is both statistically and computationally demanding.</li>
    <li>Model-based methods for inferring recombination networks exist.</li>
    <li>Within BEAST 2, the available method is Bacter, which focuses on bacterial recombination.</li>
    <li>This method will be taught in the following tutorial!</li>
  </ol>
</section>
