---
title: First Steps in BEAST 2
reveal:
    center: false
    history: true
    slideNumber: true
    theme: theme.css
---

<section class="titlepage center"
         data-background="sydney.jpg">
  <div class="title">First Steps with BEAST 2</div>
  <!--div class="subtitle">With a particular focus on Bacterial Recombination</div-->

  <div class="authors" style="margin-top:5%">Tim Vaughan<br>
  (Slides adapted from lecture given by Louis du Plessis)</div>

  <div class="institution">cEvo group, D-BSSE, The ETH Zurich</div>

  <div class="date">Taming the BEAST, 18<sup>th</sup> Feb, 2019</div>

  <div class="logos" style="margin-top:5%">
    <img data-src="Taming-the-BEAST-Down-Under-Logo.png" style="width:200px">
    <img data-src="cEvo_logo_white.png" style="width:200px">
    <img data-src="eth_logo_white.png" style="width:200px">
  </div>

</section>

<section>
  <section class="center">
    <h1>Why are we here?</h1>
  </section>

  <section>
    <h2>We all have one thing in common...</h2>

    <img data-src="beast_to_data.png" style="width:100%;margin-top:10%">

    <p style="text-align:center">We all use (or want to use) BEAST 2 to analyse our data.</p>

    <div class="fragment">
      But <i>how?</i>
    </div>

  </section>

  <section>
    <h2>Bayesian inference</h2>

    <div class="flipper" style="margin-top:10%; height:4em;">
      <div style="height:4em">
        \begin{equation*}
        P(\color{blue}{\text{model params}}|\color{darkorange}{\text{data}}) =
        \frac{P(\color{darkorange}{\text{data}}|\color{blue}{\text{model params}})P(\color{blue}{\text{model params}})}{P(\color{darkorange}{\text{data}})}
        \end{equation*}
      </div>

      <div class="fragment" style="height:4em">
        \begin{equation*}
        \text{Posterior} = \frac{\text{Likelihood}\times\text{Prior}}{\text{Model Evidence}}
        \end{equation*}
      </div>
    </div>

    <dl class="fragment">
      <dt>Prior</dt>
      <dd>Original probability for the model parameters/components. All parameters have priors, whether you specify them or not!</dd>
      <dt>Posterior</dt>
      <dd>Updated probability for the model parameters in light of the data.</dd>
      <dt>Likelihood</dt>
      <dd>Probability of data give parameters (defined by model).</dd>
      <dt>Model Evidence</dt>
      <dd>Probability for data given model (any combination of parameters): used for Bayesian model selection.</dd>
    </dl>
  </section>

  <section>
    <h2>What goes into a BEAST 2 Model?</h2>


    <div class="flipper">
    <img data-src="model_components.png" style="width:100%">
    <img data-src="model_components1.png" style="width:100%" class="fragment">
    </div>
  </section>

  <section>
    <h2>The Data</h2>

    <ul>
      <li>Typically an alignment of RNA or DNA sequences.</li>
      <li>Can also be an amino acid sequence.</li>
      <li>Often split into multiple "partitions" to allow specific sites
        to evolve at diffferent rates, e.g.
        <ul>
          <li>Coding/non-coding sites,</li>
          <li>Different codon positions (1+2 vs 3).</li>
        </ul>
      <li>Not necessarily composed of genetic sequences!</li>
    </ul>

    <div style="display:flex;justify-content:space-between">
      <img data-src="morphology.png" style="height:20%" class="fragment">
      <img data-src="linguistics.png" style="height:20%" class="fragment">
    </div>
    
  </section>

  <section>
    <h2>What goes into a BEAST 2 Model?</h2>

    <img data-src="model_components2.png" style="width:100%">
  </section>

  <section>
    <h2>The Genealogy</h2>

    <p>The fundamental genealogical structure in BEAST 2 is the rooted time tree:</p>
    <img data-src="time_tree.png" style="width:100%;margin-top:3%">

    <ul style="margin-top:3%">
      <li>This tree is a "sampled" or "reconstructed" tree.</li>
      <li>Displays relationships between sampled individuals/taxa only.</li>
      <li>In contrast to the "full" tree including unsampled individuals/taxa.</li>
    </ul>
  </section>

  <section>
    <h2>What goes into a BEAST 2 Model?</h2>

    <img data-src="model_components3.png" style="width:100%">
  </section>

  <section>
    <h2>Different population dynamics produce different phylogenetic tree shapes</h2>

    <img data-src="phylodynamics.png" style="width:90%">
  </section>

  <section>
    <h2>Demographic model</h2>

    <img data-src="birth-death_tree.png" style="width:80">

    <ul>
      <li>Describes the population (incl. individuals/species/etc.) dynamics.</li>
      <li>How does a population of organisms change through time?</li>
      <li>How does the species richness change through time?</li>
      <li>Described by the "tree prior".</li>
      <li>Usually a birth-death or coalescent model.</li>
    </ul>
  </section>
    

  <section>
    <h2>What goes into a BEAST 2 Model?</h2>

    <img data-src="model_components4.png" style="width:100%">
  </section>

  <section>
    <h2>Site model</h2>

    <img data-src="site_model.png" style="width:100%">

    <ul>
      <li> Links the genome sequences to the genealogy. </li>
      <li> We observe sequences at the tips, not their histories. </li>
      <li> Multiple substitutions at the same site means not all substitutions are observed. </li>
      <li> Site model describes rates of substitution between available characters relative to genetic distance, as well as equilibrium frequencies of characters.</li>
      <li> May also permit site-to-site rate variation. </li>
    </ul>
  </section>

  <section>
    <h2>What goes into a BEAST 2 Model?</h2>

    <img data-src="model_components5.png" style="width:100%">
  </section>

  <section>
    <h2>Clock model</h2>

    <img data-src="clock_model.png" style="width:100%">

    <ul>
      <li>Determines how quickly sequences are evolving along the tree.</li>
      <li>Different tree edges may have the same or different clock rates:
        <ul>
          <li>Strict clock models: all edges have equal rates</li>
          <li>Relaxed clock models: edges may take different rates</li>
        </ul>
      </li>
    </ul>
  </section>

  <section>
    <h2>Putting it all together...</h2>

    <img data-src="posterior_components.png" style="width:100%">

    <img data-src="phylodynamic_posterior1.png" style="width:100%">


    <div class="fragment">
    <p>Assume independence of various model components:</p>
    <img data-src="phylodynamic_posterior2.png" style="width:100%">
    </div>
  </section>

</section>

<section>
  <section class="center">
    <h1>BEAST 2 In Practice</h1>
  </section>

  <section>
    <h2>BEAST 2 Workflow</h2>

    <div class="flipper">
      <img data-src="workflow1.svg" style="width:100%">
      <img data-src="workflow2.svg" style="width:100%" class="fragment">
      <img data-src="workflow3.svg" style="width:100%" class="fragment">
      <img data-src="workflow4.svg" style="width:100%" class="fragment">
      <img data-src="workflow5.svg" style="width:100%" class="fragment">
      <img data-src="workflow6.svg" style="width:100%" class="fragment">
      <img data-src="workflow7.svg" style="width:100%" class="fragment">
      <img data-src="workflow8.svg" style="width:100%" class="fragment">
  </section>
  
<!-- BEAUti -->

  <section>
    <h2>BEAUti: <b>B</b>ayesian <b>E</b>volutionary <b>A</b>nalysis <b>Uti</b>lity </h2>

    <div style="margin-top:5%;margin-bottom:5%">
    Graphical tool for setting up a BEAST analysis.
    </div>

    <center>
      <div style="width:60%">
        <dl>
          <dt>Input</dt>
          <dd>Genetic sequence data, together with other data sources (locations, sample times, etc.)</dd>
          <dt style="margin-top:5%">Output</dt>
          <dd>Compact XML description of data, model and prior distributions.</dd>
        </dl>
      </div>
    </center>
    
  </section>

  <section>
    <h2>BEAUti: <b>B</b>ayesian <b>E</b>volutionary <b>A</b>nalysis <b>Uti</b>lity </h2>

    <center>
    <img data-src="tools_beauti.png" style="width:80%;margin-top:0">
    </center>
  </section>

  <section>
    <h2>BEAUti: <b>B</b>ayesian <b>E</b>volutionary <b>A</b>nalysis <b>Uti</b>lity </h2>

    <center>
    <img data-src="tools_beauti_XML.png" style="width:70%;margin-top:0">
    </center>
  </section>

  <section>
    <h2>BEAST 2: <b>B</b>ayesian <b>E</b>volutionary <b>A</b>nalysis by <b>S</b>ampling <b>T</b>rees</h2>

    <ul>
      <li> Performs MCMC analyses of sequences under selected sequence evolution and tree (epidemiological/speciation) model. </li>
      <li> Similar to BEAST 1.8.4/1.10 but completely separate and generally incompatable.</li>

      <li> BEAST2 and BEAST1 have a common origin, have much of the same functionality but have diverged over time. </li>

      <li> BEAST2 has a modular design that makes it easy to extend. </li>
    </ul>

    <center style="margin-top:1em">
      <div style="width:60%">
        <dl>
          <dt>Input</dt>
          <dd>XML model description file</dd>
          <dt>Output</dt>
          <dd><ul><li>(Trace) log file</li>
          <li>Tree (log) file</li></ul></dd>
        </dl>
      </div>
    </center>
  </section>

  
  <section>
    <h2>BEAST 2: <b>B</b>ayesian <b>E</b>volutionary <b>A</b>nalysis by <b>S</b>ampling <b>T</b>rees</h2>

    <center style="margin-top:-1.5em">
      <img data-src="tools_beast.png" style="width:50%">
    </center>
    
  </section>

  <section>
    <h2>BEAST 2 Packages</h2>

    <ul>
      <li>BEAST 2 is organized into a central "core" together with a large number of separate "packages".</li>
      <li>Packages can be developed by anybody - including you!</li>
      <li>Can be directly integrated into BEAST 2 and updated frequently without waiting for a
        full BEAST 2 release.</li>

      <li>Packages add new models or completely new functionality :
        <ul>
          <li>Phylogeography,</li>
          <li>bacterial ARG inference,</li>
          <li>morphological models,</li>
          <li>model selection and averaging,</li>
          <li>stochastic simulations,</li>
          <li>...</li>
        </ul>
          
      <li>Install new packages through BEAUti.</li>
    </ul>
  </section>

  <section>
    <h2>BEAST 2 Packages</h2>
    <center>
      <img data-src="tools_packages.png" style="width:100%">
    </center>
 
  </section>

  <section>
    <h2>Tracer (http://beast.community)</h2>

    <ul class="spaced">
      <li>Analyse (parameter) log files from BEAST2 runs</li>
      <li>Assess mixing, ESS, ACT, parameter correlations</li>
      <li>Provides overview of posterior parameter estimates</li>
      <li>Comparisons of several analyses.</li>
      <li>Tracer is primarily a diagnostic tool — usually want to perform
        final analyses in a statistical package like R.</li>
    </ul>

    <center style="margin-top:1em">
      <div style="width:60%">
        <dl>
          <dt>Input</dt>
          <dd>One or more log files</dd>
          <dt>Output</dt>
          <dd>Insight</dd>
        </dl>
      </div>
    </center>

  </section>

  <section>
    <h2>Tracer (http://beast.community)</h2>


    <center>
      <div class="flipper">
        <img data-src="tools_tracer_marginal.png" style="width:100%">
        <img data-src="tools_tracer_trace.png" style="width:100%" class="fragment">
        <img data-src="tools_tracer_joint_marginal.png" style="width:100%" class="fragment">
      </div>
    </center>

  </section>

  <section>
    <h2>TreeAnnotator (included with BEAST 2)</h2>

    <ul class="spaced">
      <li>Analyse trees file from BEAST2 runs.</li>
      <li>Produces single summary tree with node annotations  (including clade posterior probabilities).</li>
      <li>Positions internal nodes according to average taxon set MRCA times in trees file.</li>
      <li>Note that the MCC tree is just a heuristic summary: may produce negative edge lengths when topological uncertainty is large!</li>
    </ul>

    <center style="margin-top:1em">
      <div style="width:60%">
        <dl>
          <dt>Input</dt>
          <dd>Tree log file</dd>
          <dt>Output</dt>
          <dd>File containing annotated summary tree</dd>
        </dl>
      </div>
    </center>
  </section>

  <section>
    <h2>TreeAnnotator (included with BEAST 2)</h2>

    <center style="margin-top:5%">
    <img data-src="tools_treeannotator.png" style="width:70%">
    </center>
  </section>

  <section>
    <h2>FigTree (tree.bio.ed.ac.uk/software/figtree/)</h2>

    <ul class="spaced">
      <li> Visualise trees from BEAST2 runs.</li>
      <li> Annotate branches and nodes with probabilities and labels.</li>
      <li> Many different tree visualisation styles: circular, unrooted, etc.</li>
      <li> Allows highlighting of particular clades, colouring of edges and more.</li>
    </ul>

    <center style="margin-top:1em">
      <div style="width:60%">
        <dl>
          <dt>Input</dt>
          <dd>Tree file (eg. TreeAnnotator output)</dd>
          <dt>Output</dt>
          <dd>Tree visualisation</dd>
        </dl>
      </div>
    </center>

  </section>

  <section>
    <h2>FigTree (tree.bio.ed.ac.uk/software/figtree/)</h2>

    <center style="margin-top:-1em">
      <img data-src="tools_figtree.png" style="width:90%">
    </center>
  </section>

  <section>
    <h2>IcyTree (icytree.org)</h2>
    <ul class="spaced" style="margin-top:5%">
      <li> Similar to FigTree, but places an emphasis on quick visualisation rather than
      publication quality output.</li>
      <li> Only rooted rectangular style visualisation supported.</li>
      <li> Rudimentary support for phylogenetic networks.</li>
      <li> Web app: no installation required, just visit <a href="">icytree.org</a>.</li>
    </ul>
  </section>


  <section>
    <h2>IcyTree (icytree.org)</h2>
    <center style="margin-top:-1em">
      <img data-src="tools_icytree.png" style="width:85%">
    </center>
  </section>

  <section>
    <h2>ggtree<br>(guangchuangyu.github.io/software/ggtree/)</h2>

    <ul class="spaced" style="margin-top:5%">
      <li>R-package to visualise trees using something like Hadley Wickham's grammar of graphics (ggplot)</li>
      <li>Works with BEAST2 tree files (and many other packages)</li>
      <li>Can be easily annotate trees with other analyses in R</li>
    </ul>

    <div style="display:flex; align-items:center; justify-content:space-between;margin-top:5%">
      <img data-src="tools_ggtree1.png" style="height:200px">
      <img data-src="tools_ggtree2.png" style="height:200px">
      <img data-src="tools_ggtree3.png" style="height:200px">
    </div>
  </section>
  
<!-- BEAST -->

<!-- Tracer -->

<!-- TreeAnnotator -->

<!-- Figtree/IcyTree/ggtree -->

</section>

<section>
  <section class="center">
    <h1>Tutorial</h1>
  </section>
</section>
